tools/llama-bench/llama-bench.cpp

```diff
@@ -2134,14 +2144,32 @@ int main(int argc, char ** argv) {
             llama_memory_clear(llama_get_memory(ctx), false);
 
             if (t.n_depth > 0) {
-                if (params.progress) {
-                    fprintf(stderr, "llama-bench: benchmark %d/%zu: depth run %d/%d\n", params_idx, params_count,
-                            i + 1, params.reps);
+                bool is_cached = t.n_depth == cstate.depth;
```

Comment:
Wouldn't this need also to check if the model is the same, using the same KV type, or other parameters that may make the cache incompatible with the current test?

Reply:
`llama_state_seq_set_data` should (in theory) return error (i.e. `0`) when the state is incompatible with the current `llama_context`. I did try a few cases (different models, different KV cache types) and it seems to work as expected.

But it is a bit risky if somehow it's internal checks fail to detect an incompatibility, which can lead to invalid benches. So not sure - we can simply the logic to just reuse the state for the repetitions of the same test?